packages/react-aria-components/src/ComboBox.tsx

```diff
@@ -42,11 +42,7 @@ export interface ComboBoxRenderProps {
    * Whether the combobox is required.
    * @selector [data-required]
    */
-  isRequired: boolean,
-  /**
-   * State of the combobox.
-   */
-  state: Omit<ComboBoxState<unknown>, 'children' | 'setOpen' | 'toggle' | 'open' | 'close' | 'selectionManager' | 'setSelectedKey' | 'setFocused' | 'collection' | 'commit' | 'revert'>
```

Comment:
Decided to just remove state as a render prop from ComboBox, Select, and Tabs, which have a circular dependency on the children in order to build the state. We were previously omitting a lot of properties anyway, but now it is actually not possible to get the state before the first render.

If this comes up as a request we could potentially make a proxy object that has all of these properties, but uses getters so that the object can be passed in but the actual property values aren't available until after the render.